.github/workflows/lint-and-codecov.yml

```diff
@@ -0,0 +1,41 @@
+name: Lint & Generate CodeCov Report
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  coverage:
+    name: Lint & Generate Coverage Report
+    runs-on: ubuntu-20.04
+    container:
+      image: public.ecr.aws/m1r4u2a2/core:latest 
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Cache Bazel
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/bazel
+          key: ${{ runner.os }}-bazel-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE') }}
+          restore-keys: |
+            ${{ runner.os }}-bazel-
+
+      - name: Build `compile_commands.json` database
+        run: bazel run @hedron_compile_commands//:refresh_all
+
+      - name: Workaround for Custom Docker Image
```

Comment:
nano-nit: it might be worth a few works of a comment/description to explain why/what this is?

Reply:
I can try to figure out why. I just copy/pasted this tbh. 😅

Reply:
Good call. I've added some.